src/main/java/org/jenkinsci/plugins/ewm/restrictions/AbstractUserRestriction.java

```diff
@@ -0,0 +1,33 @@
+package org.jenkinsci.plugins.ewm.restrictions;
+
+import hudson.model.Cause;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.ewm.DiskPoolRestriction;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author Alexandru Somai
+ */
+public abstract class AbstractUserRestriction extends DiskPoolRestriction {
+
+    @Override
+    public boolean isAllowed(@Nonnull Run<?, ?> run, @Nonnull TaskListener listener) {
+        Cause.UserIdCause userIdCause = run.getCause(Cause.UserIdCause.class);
```

Comment:
I do not understand why this class enforces the user to have triggered the build. What if the build is triggered by another build? Don't we have to look "deeper", i.e. for the initial cause of the upstream build?